packages/contracts/contracts/models/StatusModel.sol

```diff
@@ -0,0 +1,14 @@
+// SPDX-License-Identifier: UNLICENSED
+
+pragma solidity 0.8.10;
+
+abstract contract StatusModel {
+    enum Status {
```

Comment:
These are states that our contract can be in so I'd call this enum `State`. Then `onlyInState` sounds better than `onlyInStatus`.